lib/view_component/base.rb

```diff
@@ -14,6 +15,11 @@ class Base < ActionView::Base
     class_attribute :content_areas, default: []
     self.content_areas = [] # default doesn't work until Rails 5.2
 
+    # Render a component collection.
+    def self.all(*args)
```

Comment:
Would it be better to  call this  `collection`?

Reply:
It was: https://github.com/github/view_component/pull/274#issuecomment-603619978

I'm fine with either, but have a slight preference for `all` as I think:

`.all(collection:` reads better than `.collection(collection:`.

Reply:
my argument would be we shouldn't need a `collection:` argument. My proposal is
```ruby
MyComponent.with_collection(@products, .....)
```

Reply:
I'm into the idea of `with_collection` and requiring the first argument to be the collection (something that can `to_ary`) and then dropping all the other special kwargs (like `as:` and `collection:`).

Reply:
And I think we could take or leave the `with_` prefix:

``` ruby
MyComponent.collection(@products, extra: foo)
```
``` ruby
MyComponent.with_collection(@products, extra: foo)
```